manual/custom_rule.md

```diff
@@ -1,18 +1,36 @@
 
 After the 2.2.1 release, xmake not only natively supports the construction of multi-language files, but also allows users to implement complex unknown file builds by custom building rules.
 
-We can extend the build support for other files by pre-setting the file suffixes supported by the rules:
+Custom build rules can have a set of file extensions associated to them using `set_extensions`.
+Once these extensions are associated to the rule a later call to `add_files` will automatically use this custom rule.
+Here is an example rule that will use Pandoc to convert markdown files added to a build target in to HTML files:
 
 ```lua
 -- Define a build rule for a markdown file
 rule("markdown")
     set_extensions(".md", ".markdown")
-    on_build_file(function (target, sourcefile)
-        os.cp(sourcefile, path.join(target:targetdir(), path.basename(sourcefile) .. ".html"))
+    on_build_file(function (target, sourcefile, opt)
```

Comment:
I prefer to provide an example of on_buildcmd_file by default in the documentation because it is simpler.

In addition, private.utils.progress is currently a private module, it should not be in the document, you can use print to display the progress.